model/task/task.go

```diff
@@ -1446,6 +1446,22 @@ func GetRecursiveDependenciesUp(tasks []Task, depCache map[string]Task) ([]Task,
 				tasksToFind = append(tasksToFind, dep.TaskId)
 			}
 		}
+		if t.IsPartOfSingleHostTaskGroup() {
+			tasksInGroup, err := FindTaskGroupFromBuild(t.BuildId, t.TaskGroup)
+			if err != nil {
+				return nil, errors.Wrapf(err, "error finding task group '%s'", t.TaskGroup)
+			}
+			if len(tasksInGroup) == 0 {
```

Comment:
It's probably fine to omit this check unless you really want to error. The loop below will just noop